From 1b13c07b176832078c6a2f6017e0683746eb5d80 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:19 -0400
Subject: [ethernet] i40e: print message for pre-production hardware

Message-id: <1409658042-22689-139-git-send-email-sassmann@redhat.com>
Patchwork-id: 90330
O-Subject: [RHEL7.1 PATCH 138/281] i40e: print message for pre-production hardware
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jesse Brandeburg <jesse.brandeburg@intel.com>

The driver and hardware are not expected to work correctly
with revision_id 0 hardware.  Don't prevent the user from
using it, but be sure to print a warning.

Change-ID: I3712d34752bfad458078a5f35dfd0aa0ae9fd20e
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 2c5fe33bdb7e1bd43f961b2850e45b4704ee0d5b)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 6cf25a6..3280255 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -8298,6 +8298,10 @@ static int i40e_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 
  i40e_verify_eeprom(pf);
 
+ /* Rev 0 hardware was never productized */
+ if (hw->revision_id < 1)
+  dev_warn(&pdev->dev, "This device is a pre-production adapter/LOM. Please be aware there may be issues with your hardware. If you are experiencing problems please contact your Intel or hardware representative who provided you with this hardware.\n");
+
  i40e_clear_pxe_mode(hw);
  err = i40e_get_capabilities(pf);
  if (err)
-- 
1.7.1